### PR TITLE
feat: trade execution ethers signer PoC

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

See https://github.com/shapeshift/web/issues/5101 for reference - exact same idea here

- Brings in base logic for using the ethers signer to sign and broadcast and return the Tx hash in executeTrade
- Brings in a stub signer, making assumptions about the fact that window.ethereum exists, the wallet is currently unlocked and is on the current chain - this is only for testing things out and since it will eventually be the responsibility of consumers to pass this once we move this out of web 

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

- tackles https://github.com/shapeshift/web/issues/5100 - doesn't tackle all swappers with EVM support yet, only 1inch/ZRX/THOR (not Li.Fi)

## Risk

None, this is effectively a dead block

## Testing

- Connect to native in the app and have your MM wallet connected to the same account 0 (this is to ensure we properly use `window.ethereum` and not hdwallet / chain-adapters to sign and broadcast
- Update `isExternalSignerEnabled` to true
- Ensure your MM wallet is unlocked and on the right chain for the trade you're trying to make
- Initiate a 1inch/ZRX/Thor trade
- Ensure MM prompts you to sign and the trade goes through as usual

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

Screenshot 2023-08-23 at 13 10 42
Screenshot 2023-08-23 at 13 10 52
Screenshot 2023-08-23 at 13 11 28
Screenshot 2023-08-23 at 13 11 36
Screenshot 2023-08-23 at 13 11 46
Screenshot 2023-08-23 at 13 11 53

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5152** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5152" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5152?utm_source=stack-comment).

File: src/lib/utils/evm.ts [hunk 1]
```diff
@@ -223,6 +224,32 @@ export const assertGetEvmChainAdapter = (chainId: ChainId | KnownChainIds): EvmC
...
+
+      data: txToSign.data,
+    }
+    // TODO(gomes): This is just a PoC but should handle switch chain logic before sending
+    return (async () => {
+      const transactionResponse = await signer.sendTransaction(ethersTx)
+      return transactionResponse.hash
+    })()
+  }
+
```
Comment (woodenfurniture) [inline src/lib/utils/evm.ts:252]: Could we take this a step further and pass in the provider? Or do you think we are better off waiting to implement it fully once it's external to the project (i can see that if we do it now we'll have a piece of code we never execute)

**Comment by woodenfurniture** [comment]
Concept proven ser. Shal we wait to put the swappers into a package before taking this to the next stage?

**Comment by woodenfurniture** [comment]
Closing as the ideas proven in this are effectively implemented in production for 1inch and cow swappers, with more to come

**Pull request closed without merging**